feat(route): add icable route

quiniapiezoelectricity opened the pull request:
<!-- 
If you have any difficulties in filling out this form, please refer to https://docs.rsshub.app/joinus/new-rss/submit-route
如果你在填写此表单时遇到任何困难，请参考 https://docs.rsshub.app/zh/joinus/new-rss/submit-route
-->

## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

<!--
Please include route starts with /, with all required and optional parameters.
Fail to comply will result in your pull request being closed automatically.
请在 `routes` 区域填写以 / 开头的完整路由地址，否则你的 PR 将会被无条件关闭。
如果路由包含在文档中列出可以完全穷举的参数（例如分类），请依次全部列出。

```routes
/some/route
/some/other/route
/dont/use/this/or/modify/it
/use/the/fenced/code/block/below
```

If your changes are not related to route, please fill in `routes` section with `NOROUTE`. Fail to comply will result in your PR being closed.
如果你的 PR 与路由无关, 请在 `routes` 区域 填写 `NOROUTE`，而不是直接删除 `routes` 区域。否则你的 PR 将会被无条件关闭。
-->

```routes
/i-cable/news
/i-cable/news/新聞資訊
/i-cable/news/港聞
/i-cable/news/財經資訊
/i-cable/news/兩岸國際
/i-cable/news/中國在線
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

TonyRL commented on lib/routes/icable/news.ts:
```diff
@@ -0,0 +1,243 @@
+                    .toArray();
+                item.video = $('article.post-detail-wrap video')
+                    .map((i, el) => ({
+                        sourceUrl: $(el).attr('data-url'),
```
Maybe I've missed something here, but I don't think placing m3u8 link in source will make it playable without aids from external JS lib.

TonyRL commented on lib/routes/icable/news.ts:
```diff
@@ -0,0 +1,243 @@
+    const category = ctx.req.param('category') ?? '新聞資訊';
+    const limit = ctx.req.query('limit') ?? 20;
+    const embed = ctx.req.query('embed') === '1';
+    const graphqlUrl = 'https://www.i-cable.com/graphql';
```
Since the site built atop WordPress. You can use the [WordPress REST API](https://developer.wordpress.org/rest-api/) instead.

For categories, you can find the category id first using `/wp-json/wp/v2/categories` with the search parameter `slug`. The id for `新聞資訊` is `37`. Then you can list recent posts with the required categories using `/wp-json/wp/v2/posts` with the search parameter `categories`, which will be `37` for `新聞資訊`. If you want plain categories, tags, post header image for each post in the posts API, you can add the `_embed` search parameter to `/wp-json/wp/v2/posts`

To adjust the no. of articles returned to the posts API, you can pass the `per_page` search parameter to `/wp-json/wp/v2/posts`.

This way, you can finish RSS generation in only 2 requests.

quiniapiezoelectricity commented:
Thanks for the review @TonyRL 

You are right, the WordPress Rest API is much better than the API I was using, I have rewritten the entire route to use the WordPress API instead.

The video/embed function had been removed since the API doesn't provide the link to the video and it would require fetching every single page to find the source, in addition to the fact that the m3u8 link might not be playable in every RSS reader (?) (ps: it works in my RSS reader of choice, but mileage may vary depends on RSS reader?) 

The category parameter now only supports category name in order for it to utilise the slug search parameter on the API .

Route is also renamed to i-cable

I believe everything is now fixed. Thanks again.

TonyRL commented on lib/routes/i-cable/news.ts:
```diff
@@ -0,0 +1,73 @@
+    const root = 'https://www.i-cable.com/wp-json/wp/v2';
+
+    const response = await cache.tryGet(`${root}/categories?slug=${category}`, async () => await got(`${root}/categories?slug=${category}`), config.cache.routeExpire, false);
+    const metadata = response.data[0];
```
`.data[0]` maybe empty if user provides an invalid input so please throw an exception if that happens

quiniapiezoelectricity commented:
@TonyRL Issue resolved. Thanks
